macros/stitch/base/stitch_bing_keyword_performance.sql

```diff
@@ -13,6 +12,84 @@ with source as (
     
 ),
 
+renamed as (
+
+    select
+    
+        "__SDC_PRIMARY_KEY" as keyword_performance_report_id,
+
+        convert_timezone('UTC', timeperiod)::timestamptz::date 
+            as campaign_date,
+            
+        accountid as account_id,
+        adgroupid as ad_group_id,
+        adgroupname as ad_group_name,
+        adid as ad_id,
+        campaignid as campaign_id,
+        campaignname as campaign_name,
+        keywordid as keyword_id,
+        
+        campaignstatus as campaign_status,
+        
+        coalesce(finalurl, '') as url,
+        
+        clicks,
+        impressions,
+        spend,
+        
+        rank() over (
+            partition by timeperiod::date 
+            order by _sdc_report_datetime desc
+            ) as rank
+
+    from source
+
+),
+
+parsed as (
+
+    select 
+    
+        keyword_performance_report_id,
+        campaign_date,
+        account_id,
+        ad_group_id,
+        ad_group_name,
+        ad_id,
+        campaign_id,
+        campaign_name,
+        campaign_status,
+        
+        url,
+        split_part(split_part(url, '//', 2), '/', 1)::varchar as url_host,
```

Comment:
This is a tough one! I don't think this works in all cases. @eogilvy12 recently had to do a very similar thing for the Segment package; take a look at her work in this PR and check out the final version in the Segment repo:

https://github.com/fishtown-analytics/segment/pull/2